extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureTaskLogs.java

```diff
@@ -151,14 +167,36 @@ private String getTaskReportsKey(String taskid)
   }
 
   @Override
-  public void killAll()
+  public void killAll() throws IOException
   {
-    throw new UnsupportedOperationException("not implemented");
+    log.info("Deleting all task logs from Azure storage location [bucket: %s    prefix: %s].",
+             config.getContainer(), config.getPrefix()
+    );
+
+    long now = timeSupplier.getAsLong();
+    killOlderThan(now);
   }
 
   @Override
-  public void killOlderThan(long timestamp)
+  public void killOlderThan(long timestamp) throws IOException
   {
-    throw new UnsupportedOperationException("not implemented");
+    log.info("Deleting all task logs from Azure storage location [bucket: '%s' prefix: '%s'] older than %s.",
+             config.getContainer(), config.getPrefix(), new Date(timestamp)
+    );
```

Comment:
same formatting nit:
```suggestion
    log.info(
        "Deleting all task logs from Azure storage location [bucket: '%s' prefix: '%s'] older than %s.",
        config.getContainer(),
        config.getPrefix(),
        new Date(timestamp)
    );
```
sorry there isn't a style rule for this, last we checked it wasn't possible to do, but it's been a while so it might be worth checking on again...

Reply:
done